### PR TITLE
Implement `Double3`, `QArgument.default` and some minor change

### PR DESCRIPTION
**Pull request by davidlatwe**
### What's changed
* Make `Number` able to set `min`, `max`
* Make `String` able to set `placeholder`
* Add attribute `default` to `QArgument` so each type can define their own default instead of `None`
* Fix `TypeError` when resetting `Number` type argument that has no *user default* (remain `None`)
* Implement `Double3` for data type like `vector`

Nice project, please let me know what you think :relaxed:

### CODE REVIEW

**Comment by mottosso** [comment]
Looks very good, welcome to merge when happy. :)

**Comment by davidlatwe** [comment]
Thanks @mottosso  😀

**Comment by davidlatwe** [comment]
I made some final tweaks:

* Add `Double3` into demo
* Try to get predefined default value from class if `default=None` is presented in keyword arguments
    If using `addArgument` without providing `default`, the change in 8f4218a will not work since `kwarg` did has `default` entry but is remain `None`. This new change fixed that.
* Polish widget alignment
    Label and reset button were aligning on top of each row for widgets that has extra heights.
    Although this provide a faster argument name look up, but for labels which stood by small widgets like `QLineEdit`, `QCheckBox` or `QSpinBox` may feel a bit crooked.

    align_on_top

    This maybe a subjective opinion, but I tried to make me happier anyway :grimacing: 

    align_on_top_if_higher

    Now the label will be aligned on top only if the widget's height is two times higher, and remain in center if not. Let me know if you prefer the original implementation and I'll take that commit out :relaxed:

File: qargparse.py [hunk 1]
```diff
@@ -91,6 +91,10 @@ def addArgument(self, name, type=None, default=None, **kwargs):
             tuple: Enum,
         }.get(type, type)
 
+        # Get default value from QArgument class if not provided
+        if default is None:
```
Comment (mottosso) [inline qargparse.py]: Hm, this seems a bit suspicious. Could this not get handled in the constructor of `Argument`, like the other `kwargs`?

**Comment by mottosso** [comment]
> Now the label will be aligned on top only if the widget's height is two times higher, and remain in center if not.

I like it, looks like it works. :) I remember struggling with this at some point actually.

Looks good, just that one note. Feel free to merge when you're happy.

File: qargparse.py (hunk 1 above)
Comment (davidlatwe) [reply qargparse.py]: Yeah, good point ! Should be fixed in a056c5c.

**Comment by davidlatwe** [comment]
Merging this ! :rocket:

**Pull request merged**